### ISSUE TITLE
Usage output should default to stdout not stderr for subcommands too

### ISSUE THREAD

**Issue by rhcarvalho** [spf13/cobra#114]
#71 fixed #31, making `command --help | grep something` work as expected, however it doesn't work for subcommands like `command subcmd --help | grep something`.

This can be reproduced by compiling the [example](https://github.com/spf13/cobra/tree/8f5946caaeeff40a98d67f60c25e89c3525038a3#example) program from the README, and:

```
$ go run main.go echo --help | grep help        # won't work
$ go run main.go echo --help 2>&1 | grep help   # hack
```

**Comment by bogem** [comment]
I can't reproduce it, It works for me

**Comment by rhcarvalho** [comment]
Thanks to `git bisect`, I identified it was fixed back in #303.
